### ISSUE TITLE
bug/fix type annotations in `partition`

### ISSUE THREAD

**Issue by nathan-chappell** [Unstructured-IO/unstructured#877]
**Describe the bug**
The type annotation for the `partition` function in `unstructured.partition.auto` is misleading.  It is currently:

```python
def partition(
    # ...
    file: Optional[IO] = None,
    # ...
): ...
```

However, this code follows the following path through the code:

* `unstructured.partition.auto.partition`
* `unstructured.partition.text.partition_text`
* `unstructured.file_utils.encoding.read_txt_file`
* `unstructured.file_utils.encoding.detect_file_encoding`
* `unstructured.partition.common.convert_to_bytes`

And upon inspection, it is clear that `unstructured.partition.common.convert_to_bytes` will throw if I pass in an instance of `StringIO`:

```python
def convert_to_bytes(
    file: Optional[Union[bytes, SpooledTemporaryFile, IO]] = None,
) -> bytes:
    if isinstance(file, bytes):
        f_bytes = file
    elif isinstance(file, SpooledTemporaryFile):
        file.seek(0)
        f_bytes = file.read()
    elif isinstance(file, BytesIO):
        f_bytes = file.getvalue()
    elif isinstance(file, (TextIOWrapper, BufferedReader)):
        with open(file.name, "rb") as f:
            f_bytes = f.read()
    else:
        raise ValueError("Invalid file-like object type")

    return f_bytes
```

**To Reproduce**

```python
from io import StringIO
from unstructured.partition.auto import partition
partition(file=StringIO('foobar'))

# unstructured/partition/common.py:204, in convert_to_bytes(file)
# ValueError: Invalid file-like object type
```

**Expected behavior**
I would expect that objects which conform to your type specifications can be used by your library.

**Desktop (please complete the following information):**
 - OS: windows 11
 - Browser chrome
 - Python version 3.9.17

**Comment by MthwRobinson** [comment]
@nathan-chappell - Thanks for flagging, will take a look at this when we can.

**Comment by MthwRobinson** [comment]
Closed by #878
